lost/src/main/java/com/mapzen/android/lost/internal/FusionEngine.java

```diff
@@ -99,33 +101,33 @@ protected void disable() {
         }
     }
 
-    private void enableGps() {
+    private void enableGps(LocationRequest request) {
         try {
             locationManager.requestLocationUpdates(GPS_PROVIDER,
-                    getRequest().getFastestInterval(),
-                    getRequest().getSmallestDisplacement(),
+                    request.getFastestInterval(),
```

Comment:
Currently all listeners will receive updates according the fastest interval and smallest displacement params from the last request in the list. While I think its fine for listeners to receive updates more frequently than requested it seems wrong to send updates less frequently because there was a more recent request with a slower interval.

Instead we should loop over all the requests and sent the lowest value among all requests for the interval. In other words we should use the _fastest_ fastest interval. In fact this is how its handled in the Android framework for setting the interval on an individual provider.

https://github.com/android/platform_frameworks_base/blob/4535e11fb7010f2b104d3f8b3954407b9f330e0f/packages/FusedLocation/src/com/android/location/fused/FusionEngine.java#L156-L197

The other interesting thing is that in the framework the minimum displacement used by the `FusionEngine` is always `0` regardless of the actual value in the location requests. I wonder if we should be doing the same thing?

https://github.com/android/platform_frameworks_base/blob/4535e11fb7010f2b104d3f8b3954407b9f330e0f/packages/FusedLocation/src/com/android/location/fused/FusionEngine.java#L129-L144